pkg/solana/chainwriter/lookups.go

```diff
@@ -210,63 +210,103 @@ func decodeBorshIntoType(data []byte, typ reflect.Type) (interface{}, error) {
 }
 
 // getSeedBytes extracts the seeds for the PDALookups.
-// It handles both AddressSeeds (which are public keys) and ValueSeeds (which are byte arrays from input args).
-func getSeedBytes(ctx context.Context, lookup PDALookups, args any, derivedTableMap map[string]map[string][]*solana.AccountMeta, reader client.Reader) ([][]byte, error) {
-	var seedBytes [][]byte
+// The return type is [][][]byte, where each element of the outer slice is
+// one combination of seeds. This handles the case where one seed can resolve
+// to multiple addresses, multiplying the combinations accordingly.
+func getSeedBytes(
```

Comment:
nit: With the new functionality, think we could maybe update the name to reflect. Maybe `getSeedBytesCombinations`